File: 0067-add-binary/0067-add-binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        int n = a.size(), m = b.size();
        int i = n-1, j = m-1, carry = 0;
        string ans = "";
        while(i >=0 && j >= 0) {
            int val = (a[i] - '0') + (b[j] - '0') + carry;
            ans += (val)%2 + '0';
            carry = val/2;
            i--; j--;
        }
        while(i >= 0) {
            int val = (a[i] - '0') + carry;
            ans += (val)%2 + '0';
            carry = val/2;
            i--;
        }
        while(j >= 0) {
            int val = (b[j] - '0') + carry;
            ans += (val)%2 + '0';
            carry = val/2;
            j--;
        }
        if(carry) {
            ans += '0' + carry;
        }
        reverse(ans.begin(), ans.end());
        return ans;
    }
};

File: 0067-add-binary/0067-add-binary.java
class Solution {
    public String addBinary(String a, String b) {
        StringBuilder ans = new StringBuilder();
        int i = a.length() - 1, j = b.length() - 1, num = 0;
        while(i >= 0 || j >= 0 || num == 1) {
            num += i>=0 ? (int)(a.charAt(i) - '0') : 0;
            num += j>=0 ? (int)(b.charAt(j) - '0') : 0;
            ans.append(num%2);
            num = num/2;
            i--;
            j--;
        }
        
        return ans.reverse().toString();
    }
}